src/compas/geometry/brep/brep.py

```diff
@@ -0,0 +1,410 @@
+from compas.geometry import Geometry
+from compas.plugins import pluggable
+
+
+@pluggable(category="factories")
+def new_brep(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_mesh(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_box(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_cylinder(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_sphere(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_cone(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+@pluggable(category="factories")
+def new_brep_from_surface(cls, *args, **kwargs):
+    raise NotImplementedError
+
+
+class BRep(Geometry):
+    """Class for Boundary Representation of geometric entities.
+
+    Attributes
+    ----------
+    vertices : list[:class:`~compas_occ.brep.BRepVertex`], read-only
+        The vertices of the BRep.
+    edges : list[:class:`~compas_occ.brep.BRepEdge`], read-only
+        The edges of the BRep.
+    loops : list[:class:`~compas_occ.brep.BRepLoop`], read-only
+        The loops of the BRep.
+    faces : list[:class:`~compas_occ.brep.BRepFace`], read-only
+        The faces of the BRep.
+    frame : :class:`~compas.geometry.Frame`, read-only
+        The local coordinate system of the BRep.
+    area : float, read-only
+        The surface area of the BRep.
+    volume : float, read-only
+        The volume of the regions contained by the BRep.
+
+    Other Attributes
+    ----------------
+    occ_shape : ``TopoDS_Shape``
+        The underlying OCC shape of the BRep.
+    type : {TopAbs_COMPOUND, TopAbs_COMPSOLID, TopAbs_SOLID, TopAbs_SHELL, TopAbs_FACE, TopAbs_WIRE, TopAbs_EDGE, TopAbs_VERTEX, TopAbs_SHAPE}, read-only
+        The type of BRep shape.
+    orientation : {TopAbs_FORWARD, TopAbs_REVERSED, TopAbs_INTERNAL, TopAbs_EXTERNAL}, read-only
```

Comment:
also this constants seem to be `occ` specific, right?

Reply:
yes they are. we could wrap this and make it more general so it applies to both/all backends...